### PR TITLE
[SPARK-35812][PYTHON] Throw ValueError if version and timestamp are used together in to_delta

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?

Throw ValueError if version and timestamp are used together in to_delta

### Why are the changes needed?
read_delta has arguments named `version` and `timestamp`, but they cannot be used together.

We should raise the proper error message when they are used together.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by Yikun** [comment]
cc @itholic @ueshin @HyukjinKwon

**Comment by HyukjinKwon** [comment]
cc @cloud-fan

File: python/pyspark/pandas/namespace.py [hunk 1]
```diff
@@ -562,6 +562,8 @@ def read_delta(
     3      13
     4      14
     """
+    if version is not None and timestamp is not None:
+        raise ValueError("version and timestamp cannot be used together.")
```
Comment (cloud-fan) [inline python/pyspark/pandas/namespace.py:566]: how about the document?

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**